src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -197,23 +221,74 @@ export class ESLPopup extends ESLToggleable {
     observer.observe(target);
 
     window.addEventListener('resize', this._deferredUpdatePosition);
-    window.addEventListener('scroll', this._deferredUpdatePosition);
+    window.addEventListener('scroll', this.onActivatorScroll, scrollOptions);
 
     this._activatorObserver = {
       unsubscribers,
       observer
     };
+
+    document.body.addEventListener('transitionstart', this._startUpdateLoop);
   }
 
+  /** Removes activator listeners and observers after hiding popup */
   protected _removeActivatorObserver(target: HTMLElement) {
     window.removeEventListener('resize', this._deferredUpdatePosition);
-    window.removeEventListener('scroll', this._deferredUpdatePosition);
+    window.removeEventListener('scroll', this.onActivatorScroll, scrollOptions);
     this._activatorObserver.observer?.disconnect();
     this._activatorObserver.observer = undefined;
     this._activatorObserver.unsubscribers?.forEach((cb) => cb());
     this._activatorObserver.unsubscribers = [];
+
+    document.body.removeEventListener('transitionstart', this._startUpdateLoop);
+  }
+
+  /**
+   * Starts loop for update position of popup.
+   * The loop ends when the position and size of the activator have not changed
+   * for the last 2 frames of the animation.
+   */
+  @bind
+  protected _startUpdateLoop() {
+    if (!this.activator || this._updateLoopID) return;
+
+    let same = 0;
+    let lastRect = new Rect();
+    const updateLoop = () => {
+      if (!this.activator) {
```

Comment:
mb I'm not quite right, but it seems to me that we will never move on to actions inside this `if`, because if `this.activator` is false, we will exit the function on line 253 (because "||" finds the first true value)